src/transformers/models/xglm/modeling_xglm.py

```diff
@@ -778,13 +781,18 @@ def forward(
 
         loss = None
         if labels is not None:
-            # shift labels and add a pad token to the end
             shift_labels = labels.new_zeros(labels.shape)
             shift_labels[:, :-1] = labels[:, 1:].clone()
             shift_labels[:, -1] = self.config.pad_token_id
```

Comment:
we can also have a seperate function for xlm, would allow not importing from loss utils